modules/network-observability-flowcollector-kafka-config.adoc

```diff
@@ -0,0 +1,27 @@
+// Module included in the following assemblies:
+
+// * configuring-operators.adoc
+
+:_content-type: PROCEDURE
+[id="network-observability-flowcollector-kafka-config_{context}"]
+
+= Configuring the Flow Collector resource with Kafka
+You can configure the `FlowCollector` resource to use Kafka. A Kafka instance needs to be running, and a Kafka topic dedicated to  {product-title} Network Observability must be created in that instance. For more information, refer to your Kafka documentation, such as link:(https://access.redhat.com/documentation/en-us/red_hat_amq/7.7/html/using_amq_streams_on_openshift/using-the-topic-operator-str)[Kafka documentation with AMQ Streams ].
+
+The following example shows how to modify the `FlowCollector` resource for {product-title} Network Observability operator to use Kafka:
+
+.Sample Kafka configuration in `FlowCollector` resource
+[id="network-observability-flowcollector-configuring-kafka-sample_{context}"]
+[source, yaml]
+----
+  deploymentModel: KAFKA                                    <1>
+  kafka:
+    address: "kafka-cluster-kafka-bootstrap.netobserv"      <2>
+    topic: network-flows                                    <3>
+    tls:
+      enable: false                                         <4>
+----
+<1> Set `spec.deploymentModel` to `KAFKA` instead of `DIRECT` to enable the Kafka deployment model.
+<2> `spec.kafka.address` refers to the Kafka bootstrap server address. You can specify a port if needed, for instance `kafka-cluster-kafka-bootstrap.netobserv:9093` for using TLS on port 9093.
+<3> `spec.kafka.topic` should match the name of a topic created in Kafka.
+<4> `spec.kafka.tls` can be used to encrypt all communications to and from Kafka with TLS or mTLS. When enabled, Kafka CA certificate must be available as a ConfigMap or a Secret, both in the namespace where the flowlogs-pipeline processor component is deployed (default: `netobserv`) and where the eBPF agents are deployed (default: `netobserv-privileged`). It must be referenced with `spec.kafka.tls.caCert`. When using mTLS, client secrets must be available in these namespaces as well (they can be generated for instance using the AMQ Streams User Operator) and referenced with `spec.kafka.tls.userCert`.
```

Comment:
we could, but I don't think we do ourselves have a good procedure for that? And we would also then have to mention how to deal with certificate rotation and so on? I just fear it would open new discussions forth and back and we're short in time ...
Also this problem is not specific to netobserv: people may already use some solutions for that (there's a bunch of third party tools, e.g. https://github.com/baloise/kube-secret-watcher ). The solution might also differ for instance for people using vaults (Hashicorp's ...)
I would just leave it up to the users for now, and we may refine that later .. sounds good?

Reply:
sure, sounds good.